### PR TITLE
Feature/rename headerkey

### PR DESCRIPTION
**Pull request by nejisama**
### Issues associated with this PR

Your PR should present related issues you want to solve.

### Solutions
You should show your solutions about the issues in your PR, including the overall solutions, details and the changes. At this time, Chinese is allowed to describe these.

### UT result
Unit Test is needed if the code is changed, your unit test should cover boundary cases, corner cases, and some exceptional cases. And you need to show the UT result.

### Benchmark
If your code involves the processing of every request, you should give the Benchmark Result.

### Code Style
+ Make sure `Goimports` has run
+ Show `Golint` result

### CODE REVIEW

File: pkg/proxy/util.go [hunk 1]
```diff
@@ -38,13 +38,13 @@ func parseProxyTimeout(ctx context.Context, timeout *Timeout, route types.Route,
 	// todo: check global timeout in request headers
 	// todo: check per try timeout in request headers
 
-	if tto, ok := headers.Get(types.HeaderTryTimeout); ok {
+	if tto, ok := headers.Get(types.MosnHeaderTryTimeout); ok {
```
Comment (taoyuanyuan) [inline pkg/proxy/util.go:41]: 这个要不要不改了

File: pkg/protocol/types.go [hunk 2]
```diff
@@ -37,32 +38,14 @@ const (
...
 func init() {
-	variable.RegisterVariable(variable.NewIndexedVariable(MosnHeaderDirection, nil, nil, variable.BasicSetter, 0))
-	variable.RegisterVariable(variable.NewIndexedVariable(MosnHeaderScheme, nil, nil, variable.BasicSetter, 0))
-	variable.RegisterVariable(variable.NewIndexedVariable(MosnHeaderHostKey, nil, nil, variable.BasicSetter, 0))
-	variable.RegisterVariable(variable.NewIndexedVariable(MosnHeaderPathKey, nil, nil, variable.BasicSetter, 0))
-	variable.RegisterVariable(variable.NewIndexedVariable(MosnHeaderQueryStringKey, nil, nil, variable.BasicSetter, 0))
-	variable.RegisterVariable(variable.NewIndexedVariable(MosnHeaderMethod, nil, nil, variable.BasicSetter, 0))
-	variable.RegisterVariable(variable.NewIndexedVariable(MosnOriginalHeaderPathKey, nil, nil, variable.BasicSetter, 0))
-	variable.RegisterVariable(variable.NewIndexedVariable(IstioHeaderHostKey, nil, nil, variable.BasicSetter, 0))
+	variable.RegisterVariable(variable.NewIndexedVariable(types.VarDirection, nil, nil, variable.BasicSetter, 0))
```
Comment (taoyuanyuan) [inline pkg/protocol/types.go:42]: 这个放到变量定义的地方吧

File: pkg/router/routers_impl.go [hunk 3]
```diff
@@ -181,11 +180,10 @@ func (ri *routersImpl) findVirtualHost(ctx context.Context) types.VirtualHost {
 		}
 		return ri.virtualHosts[ri.defaultVirtualHostIndex]
 	}
-	//we use domain in lowercase
-	key := strings.ToLower(protocol.MosnHeaderHostKey)
-	hostHeader, err := variable.GetVariableValue(ctx, key)
+	hostHeader, err := variable.GetVariableValue(ctx, types.VarHost)
```
Comment (taoyuanyuan) [inline pkg/router/routers_impl.go:183]: 看逻辑变了呢 之前是小写

Comment (nejisama) [reply pkg/router/routers_impl.go:183]: 没有变，这个是key的值，不是value，本来就是我们自己定义的，只有小写。

File: pkg/protocol/types.go (hunk 2 above)
Comment (nejisama) [reply pkg/protocol/types.go:42]: 看上去是有一部分重复了，原来有同名的Key就重复了

Comment (taoyuanyuan) [reply pkg/protocol/types.go:42]: 要么都统一都放proxy

File: pkg/types/constant.go [hunk 4]
```diff
@@ -23,25 +23,11 @@ import (
...
-	HeaderStremEnd                 = "x-mosn-endstream"
-	HeaderRPCService               = "x-mosn-rpc-service"
-	HeaderRPCMethod                = "x-mosn-rpc-method"
-	HeaderXprotocolSubProtocol     = "x-mosn-xprotocol-sub-protocol"
-	HeaderXprotocolStreamId        = "x-mosn-xprotocol-stream-id"
-	HeaderXprotocolRespStatus      = "x-mosn-xprotocol-resp-status"
-	HeaderXprotocolRespIsException = "x-mosn-xprotocol-resp-is-exception"
-	HeaderXprotocolHeartbeat       = "x-protocol-heartbeat"
+	MosnHeaderGlobalTimeout = "x-mosn-global-timeout"
+	MosnHeaderTryTimeout    = "x-mosn-try-timeout"
```
Comment (taoyuanyuan) [inline pkg/types/constant.go:29]: 要不要都统一放 variable.go 了

File: pkg/proxy/util.go (hunk 1 above)
Comment (nejisama) [reply pkg/proxy/util.go:41]: 可以吧，这里本来是想都换掉和原来做一个区分

File: pkg/types/constant.go (hunk 4 above)
Comment (nejisama) [reply pkg/types/constant.go:29]: 这个不是变量，这个就是保留在Header里的

File: pkg/protocol/types.go (hunk 2 above)
Comment (nejisama) [reply pkg/protocol/types.go:42]: 统一放在proxy里感觉不合适，protocol的适用性应该比proxy强，我把proxy里重复的删掉了

**Review by taoyuanyuan — APPROVED** [review]
LGTM
LGTM

File: pkg/router/base_rule_test.go [hunk 5]
```diff
@@ -454,7 +509,16 @@ func Test_RouteRuleImplBase_FinalizeRequestHeaders(t *testing.T) {
 				headers:     protocol.CommonHeader{"host": "xxx.default.svc.cluster.local"},
 				requestInfo: nil,
 			},
-			want: protocol.CommonHeader{"host": "xxx.default.svc.cluster.local", "authority": "www.xxx.com", "level": "1,2,3", "route": "true", "vhost": "true", "global": "true"},
+			want: &finalizeResult{
+				variables: map[string]string{
+					types.VarIstioHeaderHost: "www.xxx.com",
```
Comment (cch123) [inline pkg/router/base_rule_test.go:514]: .....

Comment (nejisama) [reply pkg/router/base_rule_test.go:514]: 这是什么问题？

**Review by taoyuanyuan — APPROVED** [review]
LGTM

**Review by cch123 — APPROVED** [review]

**Pull request merged**